### ISSUE TITLE
Implement ISO20022 replacement for ISR v11

### ISSUE THREAD

**Issue by philhug** [OCA/l10n-switzerland#302]
The Swiss ISR v11 files will be replaced with one of the following options:
* camt.053 including payment details
* camt.053 without details + separate camt.054 with payment details

Testing links (free sign up and test data):
https://isotest.postfinance.ch/

This issue is to track ideas and implementations about the ISO20022 migration.

**Comment by yvaucher** [comment]
camt import must be based on
https://github.com/OCA/bank-statement-import/tree/10.0/account_bank_statement_import_camt

**Comment by ghost** [comment]
Here's some feedback from Open Net Sàrl.

We've signed up for camt.053 including payment details with PostFinance --- first for testing in parallel V11 stream, and now in production ---, and it seems Odoo 9.0 + OCA/bank-statement-import can handle it pretty well out of the box for what I can tell:
* This is not automated: the accountant has to manually download a daily bank statement on E-Finance, and then feed it into Odoo via the Web Interface.

* It's our real world day-to-day accounting data.

* No localization involved; I'm repeating myself and in theory this should always be true when talking about international standards, but sometimes it's not the case in practice so it's worth mentioning.

* Seems account_bank_statement_import_camt by itself is able to properly eat PostFinance's camt.053s.  The first ones we received had a weird structure that seemed to contradict the [Business Rules](https://www.six-interbank-clearing.com/dam/downloads/en/standardization/iso/swiss-recommendations/business-rules.pdf) section 9.3.1, wherein we'd get exactly one C-Level with as many D-Levels as there were transactions, but now PostFinance seems to have fixed it.  In any case, I am [extending account_bank_statement_import_camt to handle that.](https://github.com/OCA/bank-statement-import/pull/102)

* PostFinance delivers a camt.053 with no transactions on a day without transactions, but [account_bank_statement_import](https://github.com/odoo/odoo/blob/10.0/addons/account_bank_statement_import/account_bank_statement_import.py) doesn't want that at all.  Apparently this is a problem with PostFinance and not with Odoo.

* ~~The reconciliation widget is able to match the banking line with the invoice using the BVR reference out of the box, which is nice.~~ (turns out it's not doing that, rather it uses bank account numbers, which causes some issues)

* We are very early adopters AFAIK. PostFinance is the first banking platform to make the leap, and we've had to talk with Support a lot.  We are currently recommending our clients to stay with V11 until everything works perfectly.

To summarize, there are some minor rough edges, but migrating from v11 to camt is already pretty feasible now.  Thoughts?

**Comment by davidwul** [comment]
thanks @lbeopennet 
Here is a few thought as we are moving our modules to v9/10
if you have a lot of "grouped transactions", you might not want all transactions in the camt053. But you will opt for the separate notification through camt054. Reconciling a bank statement and a _v11 like_ statement is a different business, especially when you have > 500 isr transactions. Having them mixed is not a good idea and source of errors. 

The camt054 is not yet parsable by the camt parser. camt054 is the "real replacement of the v11" and will be parsable thanks to your previous PR and a [minor modification](https://github.com/OCA/bank-statement-import/pull/106/files).

With postfinance, you can fully automate the process [with the fds module ](https://github.com/OCA/l10n-switzerland/pull/323)that will download and import the camt053 automatically. If you're with a bank, full automation could also become true in some future with the [ebics initiative](https://www.six-interbank-clearing.com/dam/downloads/fr/clearit/57/ebics.pdf). So the real world of accounting data is getting better and better! 

To be able to import camt notification with images you will need also [the new tar.gz parser wich is in PR](https://github.com/OCA/l10n-switzerland/pull/324)

We are also working on improving the cam053 to have more details. It appears the some banks are forwarding a lot more details on payee address and iban. We [have done a PR](https://github.com/OCA/bank-statement-import/pull/105) on the bank-statement-import branch.

Sticking with v11 is definitely the choice for now, but if I'm right v11 is available only this year and will be retired in January 2018

**Comment by ghost** [comment]
@yvaucher, (or someone else) why is [the matching in base_transaction_id](https://github.com/OCA/bank-statement-reconcile/blob/10.0/base_transaction_id/models/account_bank_statement_line.py#L14) done by comparing `account.move.line.transaction_ref` with `account.bank.statement.name` rather than `account.bank.statement.ref`? On our setup, it seems that both contain ISR numbers in the same format. Maybe we could search both? The [V11 importer](https://github.com/OCA/l10n-switzerland/blob/10.0/l10n_ch_payment_slip/wizard/bvr_import.py#L160) did put the ISR number in `name`, but the Camt parser doesn't. Resolution is still easy thanks to the IBAN number matcher.

**Comment by eLBati** [comment]
@lbeopennet @davidwul thanks for your work.
What is the relation between https://github.com/OCA/bank-statement-import/pull/102 and https://github.com/OCA/bank-statement-import/pull/105?
One depends on the other? Are they overlapping? Independent?
Thanks

**Comment by ghost** [comment]
Completely independent, it's not the same "details" we are talking about.

**Comment by yvaucher** [comment]
@lbeopennet I understand you mean `account.bank.statement.line.ref` and `account.bank.statement.line.name`

This was done here
https://github.com/OCA/l10n-switzerland/commit/4be015fc4e2502889b14bbbf7170fa7cce8e2555

We had a change to have the invoice number and the ISR number. So it is in name that the ISR should be written.

Now according to field names: 

name => Label (in 9.0 it was Memo)
ref => Reference

It is surely unclear what odoo means which each fields.

What is certain is that the official module use name to find reconciliation proposition:
https://github.com/odoo/odoo/blob/9.0/addons/account/models/account_bank_statement.py#L628

When we did this change the official code was probably this one:

https://github.com/odoo/odoo/commit/5b4404f628ac2e4a8d8c0cd48a791ab7689ace68#diff-86c8da67df78eedb5113360f754e1f75L605

So we wanted to be closer as possible as official `account` module.

**Comment by davidwul** [comment]
@yvaucher thanks for the explanation.
What is best to change, 
- the way reconciliation proposition are done (and include subsequently ref after name)
- develop specific camt version in l10n_ch so in case of referenced credit (ISR/DD/LSV) _reference_ is imported in _name_? 

If the second option then this [PR would be really useful](https://github.com/OCA/bank-statement-import/pull/107).

**Comment by philhug** [comment]
@davidwul @yvaucher is there a PR for the name/ref matching issue?

**Comment by peter2018abc** [comment]
is importing of camt.054 files from PostFinance possible in odoo11?
